lib/embulk/input/marketo/lead.rb

```diff
@@ -42,35 +42,38 @@ def self.generate_columns(metadata)
         end
 
         def run
-          count = 0
           from_datetime = task[:from_datetime]
-          to_datetime = task[:to_datetime]
+          to_datetime = task[:to_datetime] || Time.now
+          ranges = task_target(from_datetime, to_datetime, task[:workers], index)
```

Comment:
In this case, same time ranges can be generated by run method, you know. But run method is executed on each worker thread. So, run method should not control (or divide) a whole of task. It should execute tasks.